components/mime.rst

```diff
@@ -0,0 +1,541 @@
+    $email = (new Email())
+        ->from('fabien@symfony.com')
+        // ...
+    ;
+    $serializedEmail = $email->toString();
+
+    // later, recreate the original message to actually send it
+    $message = new RawMessage($serializedEmail);
+
+.. _mime-component-twig-integration:
+
+Twig Integration
+----------------
+
+The Mime component integrates with the :doc:`Twig template engine </templating>`
+to provide advanced features such as CSS style inlining and support for HTML/CSS
+frameworks to create complex HTML email messages.
+
+Rendering Email Contents with Twig
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you define the contents of your email in Twig templates, use the
+:class:`Symfony\\Bridge\\Twig\\Mime\\TemplatedEmail` class. This class extends
+from the :class:`Symfony\\Component\\Mime\\Email` class explained above and adds
+some utility methods for Twig templates::
+
+    use Symfony\Bridge\Twig\Mime\TemplatedEmail;
+
+    $email = (new TemplatedEmail())
+        ->from('fabien@symfony.com')
+        ->fo('foo@example.com')
+        // ...
+
+        // this method defines the path of the Twig template to render
+        ->htmlTemplate('messages/user/signup.html.twig')
+
+        // this method defines the parameters (name => value) passed to templates
+        ->context([
+            'expiration_date' => new \DateTime('+7 days'),
+            'username' => 'foo',
+        ])
+    ;
+
+Once the email object has been created, you must `set up Twig`_ to define where
+templates are located and then, use the
+:class:`Symfony\\Bridge\\Twig\\Mime\\BodyRenderer` class to render the template
+and update the email message contents with the results. All this is done
+automatically when using the component inside a Symfony application::
+
+    // ...
+    use Symfony\Bridge\Twig\Mime\BodyRenderer;
+    use Twig\Environment;
+    use Twig\Loader\FilesystemLoader;
+
+    // when using the Mime component inside a full-stack Symfony application, you
+    // don't need to do this Twig setup. You only have to inject the 'twig' service
+    $templateLoader = new FilesystemLoader(__DIR__.'/templates');
+    $twig = new Environment($templateLoader);
+
+    $renderer = new BodyRenderer($twig);
+    // this updates the $email object contents with the result of rendering
+    // the template defined earlier with the given context
+    $renderer->render($email);
+
+The last step is to create the Twig template used to render the contents:
+
+.. code-block:: html+twig
+
+    <h1>Welcome {{ username }}!</h1>
+
+    <p>You signed up to our site using the following email:</p>
+    <p><code>{{ email.to }}</code></p>
+
+    <p><a href="{{ url('...') }}">Click here to activate your account</a></p>
+
+The Twig template has access to any of the parameters passed in the ``context``
+method of the ``TemplatedEmail`` class and also to a special variable called
+``email`` which gives access to any of the email message properties.
```

Comment:
I think you need to mention the Wrapped class here as the methods are slightly different from the regular Email class.